app/ui-utils/client/lib/renderMessageBody.js

```diff
@@ -32,4 +33,4 @@ export const renderMessageBody = mem((message) => {
 
 	return (Array.isArray(tokens) ? tokens.reverse() : [])
 		.reduce((html, { token, text }) => html.replace(token, () => text), html);
-}, 500, ({ _id, _updatedAt }) => _id && _updatedAt && _id + _updatedAt);
+}, 500, (message) => md5(JSON.stringify(message)));
```

Comment:
instead  of insert a new lib and hash all the message, what do you think about just use your fields here? we inserted that cache to skip some computations (usually, we just need update a message if the `_updatedAt ` changes), believe there are some fields we dont want to invalidate the message...

Reply:
@ggazzo imho by using additional fields for example `translations` in this case would add complexity to the cache key and will not have any difference since the cache will be invalidated after 500ms. At least i cannot imagine where this add additional keys will be beneficial.

Reply:
If your concern lies around adding a new lib, then how about [cypto](https://nodejs.org/docs/v0.6.18/api/crypto.html#crypto_crypto_createhash_algorithm)?